src/libraries/System.Runtime/tests/System/Reflection/ReflectionCacheTests.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    [SkipOnMono("Mono doesn't have a reflection cache")]
+    public class ReflectionCacheTests
+    {
+        [Fact]
+        public void GetMethod_MultipleCalls_SameObjects()
+        {
+            MethodInfo mi1 = typeof(ReflectionCacheTests).GetMethod(nameof(GetMethod_MultipleCalls_SameObjects));
+            Assert.NotNull(mi1);
+
+            MethodInfo mi2 = typeof(ReflectionCacheTests).GetMethod(nameof(GetMethod_MultipleCalls_SameObjects));
+            Assert.NotNull(mi2);
+
+            Assert.Same(mi1, mi2);
+        }
+
+        [Fact]
+        public void GetMethod_MultipleCalls_ClearCache_DifferentObjects()
```

Comment:
this test will fail in mono until https://github.com/dotnet/runtime/issues/50978 is done